UpdatedImageProcessing/Classifiers/classify_color.py

```diff
@@ -0,0 +1,220 @@
+import sys
+from os import walk
+import cv2
+import numpy as np
+
+try:
+	from PIL import Image
+except ImportError:
+	import Image
+
+class ColorClassifier(object):
+	"""
+	Step 4: Autonomously classify shape color and alphanumeric color.
+	"""
+
+	def __init__(self, img=None):
+		"""
+		Constructor.
+
+		:param img:		Opened PIL (Python Image Library) image.
+		:type img:		PIL.Image.Image
+
+		:return:		Nothing.
+		"""
+		if img != None:
+			self.img = img
+
+		self.colors = {
+			"white": [255, 255, 255, 255],
+			"black": [0, 0, 0, 255],
+			"gray": [128, 128, 128, 255],
+			"red": [255, 0, 0, 255],
+			"blue": [0, 0, 255, 255],
+			"green": [0, 255, 0, 255],
+			"yellow": [255, 255, 0, 255],
+			"purple": [128, 0, 128, 255],
+			"brown": [165, 42, 42, 255],
+			"orange": [255, 165, 0, 255]
+		}
+
+	def get_color(self, img=None):
+		"""
+		Gets the color of the background and text inside a target.
+
+		If no img is passed to get_color, then this function assumes to use the 
+		PIL image passed in the constructor.
+
+		:param img:		optional. Opened PIL image.
+		:type img:		PIL object.
+
+		:return:		Two element list, where the first element is the shape
+						color, and the second element is the text color. Ex:
+							["red", "green"]
+						or, simply:
+							[<background color>, <text color>]
+		"""
+		passed = Image.open(self.img)
+
+		# convert pil to opencv2
+		img = np.array(passed)
+		img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+
+		Z = img.reshape((-1, 3))
+
+		# convert to np.float32
+		Z = np.float32(Z)
+
+		# define criteria, number of clusters(K) and apply kmeans()
+		criteria = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 10, 1.0)
+		K = 3
+		ret, label, center = cv2.kmeans(Z, K, None, criteria, 10, cv2.KMEANS_RANDOM_CENTERS)
+
+		# now convert back into uint8, and make original image
+		center = np.uint8(center)
+		res = center[label.flatten()]
+		res2 = res.reshape((img.shape)) # actual image
+
+		# grab colors from image
+		res3 = cv2.cvtColor(res2, cv2.COLOR_RGB2BGR)
+		colors_from_img = set( tuple(v) for m2d in res3 for v in m2d )
```

Comment:
This is a very confusing for loop in a for loop. Please use better variable names.

Reply:
Done in https://github.com/FlintHill/SUAS-Competition/commit/06850a66a2687093624a87c4dd4d3d7619012d1c

Reply:
It doesn't appear to have changed?

Reply:
My bad. I thought I saved when I committed in December. Fixed in commit (line number included): https://github.com/FlintHill/SUAS-Competition/commit/47ca8bf17027cd022599e28693fadba63a6cc7e1#diff-4f24dac54a4625fc46c8e14f9b6b3158L77